### DISCUSSION TITLE
[New] Proposition d'ajout de pratique "Valider son applicatif avec une bande passante limitée"

### DISCUSSION THREAD

**Discussion by DocRoms** [cnumr/best-practices#36] [Bonnes pratiques] [answered]
### Bonne pratique : Valider son applicatif avec une bande passante limitée
Les outils de développeur des navigateurs (et certains sites d'analyses statique) permettent depuis de nombreuses années de brider sa connexion pour tester le rendu de son applicatif avec un débit réduit. 

Cette pratique est idéale, vu que l'on se rend compte réellement des méfaits de la lourdeur du web, on observe que la sobriété et que les bonnes pratiques de performance web aide beaucoup à améliorer l'expérience utilisateur sur des débits réduits. 

En validant que son applicatif répond rapidement en 2G, ou en 3G , on se rend garant du fait que l'on a agit de manière positive sur son poids à travers le réseau, que l'on agit également sur le côté `inclusif` du web, en rendant disponible son applicatif à des personnes n'ayant pas un débit optimal. On observe également les optimisations mises en place qui ne sont pas encore parfaites, et ce qu'il nous reste à modifier pour rendre l'application disponible sur des connexions plus faible. 

### Régle de validation proposée : 
 - L'application doit être fonctionnelle avec une bande passante limitée (3G)

**Comment by ACTLEM** [comment]
Je suis à 100% pour mentionner l'utilisation de ces outils. Par contre, est-ce vraiment une pratique d'écoconception ou plutôt un moyen de vérification (comme peut l'être l'analyse du nombre de requêtes, la taille du DOM etc...)?
J'avais pensé à mettre en avant quelques trucs et astuces et notamment celui-là. Pour moi, cela doit plus aller dans une rubrique "Tips" que bonnes pratiques d'écoconception web.

**Reply by DocRoms** [reply]
Tu as raison, et c'est faisable : on peut valider que son application fonctionne en bas débit :)

**Reply by ACTLEM** [reply]
Comme le propose Florine, à voir si on n'ajouterait pas une page  Mesure (Monitoring). Même si beaucoup d'entre nous le font, je pense que cela pourrait être un "pense bête" sympa, en plus c'est ce genre de page qui peut être rendue publique (quelque soit le moyen, sur EcoIndex, collectif.greenit ou via un autre repo public du CNumR pour qu'il vive bien) sans problème de licence.

**Comment by florinesueur** [comment]
Les systèmes ou procédures de vérification/validation ne sont-ils pas des bonnes pratiques en soi, comme peut l'être [[BP_031] "Valider les pages auprès du W3C"](https://github.com/cnumr/best-practices/blob/main/chapters/BP_031_fr.md) ? En ce sens, est-ce qu'il est intéressant de les comptabiliser au score d'évaluation de la maturité ? Ici, il me semble intéressant de scorer via un temps maximum de chargement pour un débit réduit donné qui est aussi un bon moyen de mesurer l'évolution. Alors, peut-être ajouter une section de mesure (monitoring) et de suivi des pratiques mises en place à ajouter ?

**Reply by ACTLEM** [reply]
Oui, avoir une section "mesure" afin de vérifier que les mises en place de bonnes pratiques d'écoconception apporte une évolution positive est une bonne idée. Je préfère d'ailleurs cela qu'à ce que j'avais proposé ("Trucs et astuces") car cela entre vraiment dans la démarche d'écoconception (mesure performance environnementale, amélioration continue).

**Comment by ACTLEM** [comment, marked as answer]
https://github.com/cnumr/best-practices/issues/138
